lib/auth/login_log.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Auth.LoginLog do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Auth.Repo
+
+  schema "login_logs" do
+    field :email, :string
+    field :ip_address, :string
+    field :auth_provider, :string
+    belongs_to :person, Auth.Person
```

Comment:
A login attempt cannot `belongs_to` a `:person` if the `:email` address does not exist in the People table.
This is an unnecessary constraint. 
If someone attempts to login with the _incorrect_ `:email` we still want to record the _attempt_. 

On the other hand, if the `:person` does not exist then it will count as _registration_,
so technically there won't be a "login attempt" at all, it will just prompt them to set a (`new`) password.
Still the `belongs_to` doesn't _need_ to be here. We can just use `person_id` and have the two tables decoupled. We can still perform a SELECT+JOIN to show the person all the logins.

Reply:
Makes sense. I've created the Ecto association to make it easier to link to the log to the person but as you said it might not be possible when the person is using a wrong email or when the record doesn't exist yet.